### PR TITLE
add actkbd; nixos: add support for volume control with media keys

### PR DESCRIPTION
**Pull request by oxij**

### CODE REVIEW

File: nixos/modules/services/hardware/actkbd.nix [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+  config = mkIf cfg.enable {
+
+    services.udev.packages = lib.singleton (pkgs.writeTextFile {
+      name = "actkbd-udev-rules";
+      destination = "/etc/udev/rules.d/61-actkbd.rules";
+      text = ''
+        ACTION=="add", SUBSYSTEM=="input", KERNEL=="event[0-9]*", ENV{ID_INPUT_KEY}=="1", TAG+="systemd", ENV{SYSTEMD_WANTS}+="actkbd@$env{DEVNAME}.service"
+      '';
+    });
```
Comment (edolstra) [inline nixos/modules/services/hardware/actkbd.nix]: Can't this be done using `services.udev.extraRules`?

Comment (oxij) [inline nixos/modules/services/hardware/actkbd.nix]: Nope. `ENV{ID_INPUT_KEY}` fields are not set before `60-<something>` in udev rules order and `extraRules` are `10-<something>`.

Comment (edolstra) [inline nixos/modules/services/hardware/actkbd.nix]: Ah okay. I guess we should add an option for late rules...

**Comment by oxij** [comment]
Changed the patch a little bit to workaround #7488

**Comment by cstrahan** [comment]
This looks pretty nice :)

**Comment by lethalman** [comment]
Haven't tested the service, however the package passes on travis. Also the udev part can be improved later in general. Merging thanks. I can revert in case it breaks nixos evaluation.

**Pull request merged**